### PR TITLE
Fix JMX exporter quoted property values captured incorrectly

### LINKED ISSUE #195: JMX exporter: quoted property values captured incorrectly
Object names with quoted property values are being incorrectly captured by regular expression defined in the JMX exporter configuration.

**Current behavior**
The quotes and the value enclosed by the quotes are being captured.

**Expected behavior**
Only the value enclosed by quotes should be captured.

Example of Prometheus metric after `name` object name property has been converted to `processor` label:

```
org_apache_camel_ExternalRedeliveries{context="MyCamel",processor="\"log1\"",type="processors",} 0.0
```
See file with more examples: [metrics.txt](https://github.com/fabric8io-images/s2i/files/2599835/metrics.txt)

### PR DESCRIPTION
**Pull request by alexkieling**
Fix #195

### CODE REVIEW

**Comment by alexkieling** [comment]
@rhuss I wasn't able to use fish-pepper because it failed to install in my machine

File: java/blocks/jmxexporter/fp-files/prometheus-config.yml [hunk 1]
```diff
@@ -233,119 +233,119 @@ rules:
...
+  - pattern: 'org.apache.camel<context=([^,]+), type=processors, name="([^"]+)"><>TotalProcessingTime'
     name: org.apache.camel.TotalProcessingTime
     help: Total Processing Time
     labels:
         context: $1
         processor: $2
         type: processors
     type: GAUGE
-  - pattern: 'org.apache.camel<context=([^,]+), type=processors, name=([^,]+)><>InflightExchanges'
+  - pattern: 'org.apache.camel<context=([^,]+), type=processors, name="([^"]+)"><>InflightExchanges'
```
[the remark this answers is not in the document]
Comment (alexkieling) [reply java/blocks/jmxexporter/fp-files/prometheus-config.yml]: The regex seems to be working fine. Try it with a demo app or on https://regex-golang.appspot.com/assets/html/index.html

All objectNames that start with `org.apache.camel` and contain a `name` attribute seem to specify the value enclosed in double quotes. The regex I used expects to find double quotes but we could change it to support both cases if you think it's better.

[the remark this answers is not in the document]
Comment (alexkieling) [reply java/blocks/jmxexporter/fp-files/prometheus-config.yml]: I've just had a look at the [JMX exporter example configurations](https://github.com/prometheus/jmx_exporter/tree/master/example_configs) and it looks like other Java apps have similar issues, e.g., Artemis and Tomcat. They also enclose some capturing groups in double quotes.

**Comment by vorburger** [comment]
@rhuss is this one good to go for you? 

@alexkieling I don't suppose it would be [possible to add a test.sh for this](https://github.com/fabric8io-images/s2i/blob/master/test.sh)? Or I could merge it just trusting you...

**Comment by alexkieling** [comment]
@vorburger Do you mean adding a new test script or changing the existing one? We could hit the /metrics endpoint and look for labels with duplicated quotes.

**Comment by alexkieling** [comment]
OK, I'll do that

**Comment by alexkieling** [comment]
@vorburger fish-pepper installation failed on Fedora 28: https://github.com/fabric8io-images/fish-pepper/issues/13 I'm currently unable to run `test.sh`.

**Comment by alexkieling** [comment]
@vorburger `test.sh` is currently testing the `s2i-java-binary-example`, which doesn't expose the Camel metrics I fixed.

**Comment by vorburger** [comment]
> @vorburger `test.sh` is currently testing the `s2i-java-binary-example` image, which doesn't expose the Camel metrics I fixed.

Oh, I see; I hadn't even paid close enough attention to notice that this was Camel specific. Tell you what - let me finally merge this as is (thanks for your patience!). I've opened #200 to follow-up, if that's something you would be motivated to help this project with?

**Pull request merged**